testing/e2e/agent_install_test.go

```diff
@@ -0,0 +1,362 @@
+	err = copyPath(suite.binaryPath, fleetPath)
+	suite.Require().NoError(err)
+
+	// link elastic-agent to the actual binary
+	if agentLink != "" {
+		suite.Require().NotEmpty(agentSrc, "agent link src undetected")
+		suite.Require().NotEmpty(agentDst, "agent link dst undetected")
+		err = os.Symlink(agentSrc, agentDst)
+		suite.Require().NoError(err)
+		suite.agentPath = agentDst // replace path if we are using a link
+	}
+}
+
+func (suite *AgentInstallSuite) TearDownSuite() {
+	if suite.downloadPath != "" {
+		err := os.RemoveAll(suite.downloadPath)
+		suite.Require().NoErrorf(err, "failed to remove download from %s", suite.downloadPath)
+
+		// FIXME work around for needing to run sudo elastic-agent install
+		err = exec.Command("sudo", "rm", "-rf", suite.downloadPath).Run()
+		if err != nil {
+			suite.T().Logf("unable to remove %q: %v", suite.downloadPath, err)
+		}
+	}
+}
+
+func (suite *AgentInstallSuite) SetupTest() {
+	if suite.installDetected {
+		suite.T().Skip("elastic-agent install detected, skipping test.")
+	}
+
+	portFree := suite.IsFleetServerPortFree()
+	suite.Require().True(portFree, "port 8220 must not be in use for test to start")
+}
+
+func (suite *AgentInstallSuite) TearDownTest() {
+	if suite.T().Skipped() {
+		return
+	}
+	path, err := exec.LookPath(agentName)
+	if err != nil {
+		suite.T().Logf("unable to detect elastic-agent install on test tear-down: %s", err)
+		return
+	}
+	err = exec.Command("sudo", path, "uninstall", "--force").Run()
+	suite.Require().NoError(err, "elastic-agent uninstall failed.")
+}
+
+func (suite *AgentInstallSuite) TestHTTP() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*3)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, "sudo", suite.agentPath, "install",
+		"--fleet-server-es=http://"+suite.esHosts,
+		"--fleet-server-service-token="+suite.serviceToken,
+		"--fleet-server-insecure-http=true",
+		"--fleet-server-host=0.0.0.0",
+		"--fleet-server-policy=fleet-server-policy",
+		"--non-interactive")
+	cmd.Env = []string{"GOCOVERDIR=" + suite.coverPath} // TODO Check if this env var will be passed by the agent to fleet-server
+	cmd.Dir = filepath.Dir(suite.agentPath)
+
+	output, err := cmd.CombinedOutput()
+	suite.Require().NoErrorf(err, "elastic-agent install failed. command: %s, exit_code: %d, output: %s", cmd.String(), cmd.ProcessState.ExitCode(), string(output))
+
+	suite.FleetServerStatusOK(ctx, "http://localhost:8220")
+}
+
+func (suite *AgentInstallSuite) TestWithSecretFiles() {
+	dir := suite.T().TempDir()
+	err := os.WriteFile(filepath.Join(dir, "service-token"), []byte(suite.serviceToken), 0600)
+	suite.Require().NoError(err)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute*3)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, "sudo", suite.agentPath, "install",
+		"--url=https://localhost:8200",
+		"--certificate-authorities="+filepath.Join(suite.certPath, "e2e-test-ca.crt"),
+		"--fleet-server-es=http://"+suite.esHosts,
```

Comment:
Nit. `esHost` is described as a comma separated list, will this work with more than one host?

Reply:
probably not